internal/client/discord/discord_client.go

```diff
@@ -0,0 +1,46 @@
+package discord
+
+import (
+	"github.com/bwmarrin/discordgo"
+	"haracho-go/internal/client"
+	"haracho-go/internal/logger"
+	"haracho-go/internal/service"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+type Client struct {
+	Token   string
+	session *discordgo.Session
+}
```

Comment:
```suggestion
type Client struct {
	Token      string
	session    *discordgo.Sessionc
	Collection *client.CommandCollection
}
```
`CommandCollection` を外部から入れてもらう方が制御が効く